### ISSUE TITLE
Change startBlockNumber to canonicalBlock

### ISSUE THREAD

**Issue by jo-tud** [ConsenSys/EthOn#85]
Canonical Block is a better term for the first Block that is part of a Blockchain (especially useful for forked chains). Also it should not be a data property but an object property pointing to the actual Block.

**Comment by sunnya97** [comment]
A blockchain's beginning should be defined by its genesis block, not the block at which it forked.  I don't see there being an issue with a block being in multiple Blockchains.  What's more useful for defining a specific blockchain rather is the most recent block, because once that is defined, getting the rest of the blocks is trivial as you can just work backwards through the chain of parents.

**Comment by webthethird** [comment]
I would agree with @sunnya97 's comment, despite the multiple forks, I would still consider the genesis block to be the beginning of the Ethereum blockchain (as well as ETC...)

**Comment by jo-tud** [comment]
This really is a tricky issue.
Of course there is a Genesis Block for all blockchains, which is the first block of the common past. But with each protocol change a new `segment` in the blockchain is started and a new canonical block is created.

The various forks of a blockchain form a tree. And each segment has a canonical block at its root.
This graphic illustrates that fact:

![fork modeling](https://user-images.githubusercontent.com/1543640/28789006-961bf8fa-7623-11e7-8bb6-0959a0dc5171.png)

In the above image - what is the blockchain? It's more a block tree! And the segments are identified by canonical blocks (those where a fork happens). For the non forking segment, the canonical block has no meaning. Only for the fork.
